### ISSUE TITLE
Highlight current filter in right drawer. If user changes any sort, tag, list settings, remove highlight.

### ISSUE THREAD

**Issue by Alphalpha007** [mpcjanssen/simpletask-android#497]

**Comment by mpcjanssen** [comment]
How about showing the filter in the Title bar? Easier to implement and doesn't require a swipe.

**Comment by Alphalpha007** [comment]
Are you saying you'll remove right drawer?

**Comment by mpcjanssen** [comment]
No of course not :smile: but when you select a saved filter, the name of the filter could be displayed in the title.

**Comment by smichel17** [comment]
@alphalpha007 Not proposing it for this issue, but how would you feel if we put the active filter in the titlebar, and if you tapped on the name of the active filter it would show a drop down menu for switching between filters?

**Comment by Alphalpha007** [comment]
@mpcjanssen alright, makes sense.

@smichel17, I'm okay with that. But then you have to figure out how to accommodate the add, import and export options that's currently there in the right drawer.

**Comment by smichel17** [comment]
@alphalpha007 Indeed. This is also in the context of a bunch of other changes. Lots of variables and lots of options. Here I'm just thinking about the one part of where to put the interface for changing filters.

What about if the filters were tabs?

**Comment by Alphalpha007** [comment]
@smichel17 I wouldn't prefer tabs for the simple reason that when the number of filters is large it's an inconvenience.

**Comment by smichel17** [comment]
@Alphalpha007 I'm imagining in the future, sorting will happen in a separate stage from filtering; filtering will only be about controlling *which* tasks get shown. In that future, you only have 3 filters, so maybe it's actually nicer to have them in tabs.

**Comment by Alphalpha007** [comment]
What are these 3 filters?

**Comment by smichel17** [comment]
https://github.com/mpcjanssen/simpletask-android/issues/458#issuecomment-232572759

**Comment by Alphalpha007** [comment]
I see. But I have already created two filters after I made that comment, and really useful ones too. Just saying that a scrollable dropdown would be much cleaner.

**Comment by smichel17** [comment]
tl;dr, we need to make a pretty major change to the way Simpletask's internals think about saved filters in order to implement this the right way. We could implement it a wrong, hack-y way, but I have no desire to inflict that on @mpcjanssen or myself

---
This is a hard issue to implement right now because the code doesn't have any concept of a "saved-filters-as-active-filter". There's just one filter, the active filter. A saved filter is actually a shortcut that sets the active filter to certain values. So if you open a saved filter, right at that moment the code knows that you have that filter open, but if you use the quick filter drawer to change that filter, the code sees that you're looking at a different filter, not the saved one.

If you then undo what you did in the filter drawer to get back to the saved filter, it still doesn't know that the saved filter is open because there's not currently a way of comparing the active filter to the saved filter (it's possible to create one, but it would be a real pain to code and would be very vulnerable to bugs). So it's possible to set the title or apply a highlight immediately when you select the saved filter, but not when returning to it.

That's all well and good, except that much of the UI changes (without saving the previous state) any time you select tasks or open a drawer -- ie any time you do anything. So we could do the hack-y way of writing a function to compare the active filter to each saved filter and see if they're the same, but the right way to write this feature is to add the concept of "saved-filter-as-active-filter", so that the part of the code that sets up the UI won't have to check the active filter against the saved filter each time or keep track of previous states, which also gets ugly quickly, compared to just looking at the current state and creating the UI accordingly.

The problem is, "saved-filter-as-active-filter" doesn't make sense with the way Simpletask currently works. Say you have one filter that shows things with @a and one with @b. You load filter A, then open the quick filter drawer and uncheck @a and check @b. According to the hack-y way we'd have to do things right now, we'd still mark filter A as the active filter, even though filter B is currently shown.
@ddisanti **Or, say you selected filter A, opened the quick filter drawer, and added filtering for +c. I think you'd still expect filter A to be highlighted,** but you couldn't do that and avoid the bug above.

@Alphalpha007 I know that you're aware of these issues; I'm writing this mostly for other people who come to read this issue. We could of course use your suggestion and only show the highlight/title right when the filter is selected, but per the bold sentence above, I think the feature would feel incomplete and I'd prefer to implement it fully, the right way, instead of in an incomplete and/or hack-y way.
